### ISSUE TITLE
Replace go-cron with crond from the already included BusyBox

### ISSUE THREAD

**Issue by jlelse** [djmaze/resticker#11]
Can we get rid of go-cron and use the BusyBox cron features directly? (like described here: https://stackoverflow.com/a/37016878)

I don't really trust go-cron. Although the restic container (1.0.0) was running for 3 days, today I noticed it didn't do any backups since then, although the logs said "new cron: 0 0 2 * * *".

I will try to create a PR soon.

**Comment by y3lousso** [comment]
+1 in the mazzolino/restic:1.0, cron doesn't work.

**Comment by y3lousso** [comment]
So I tried 2 things : 

### Test go-cron : 
`docker exec container_name go-cron "* * * * * *" /bin/bash -c "echo 1"`
No success, go-cron command doesn't work.
Either the setup of go-cron is wrong, or go-cron doesn't work.

### Replace go-cron by cron from BusyBox : 

1. Removed go-cron from the Dockerfile
2. Replace go-cron by cron (BusyBox) in the entrypoint : 
`exec go-cron "$BACKUP_CRON" /usr/local/bin/backup`
becomes
`echo "Setup backup cron job with cron expression BACKUP_CRON: ${BACKUP_CRON}"`
  `echo "${BACKUP_CRON} /backup >> /var/log/cron.log 2>&1" > /var/spool/cron/crontabs/root`
  `# Make sure the file exists before we start tail`
  `touch /var/log/cron.log`
  `# Start the cron daemon`
  `crond`

Still no success, the container crashes without giving any proper log : 
`container_name with code 0`

If someone has a better (at least working) solution, please share. :+1:

**Comment by jlelse** [comment]
@y3lousso Looks great! :+1:

**Comment by jlelse** [comment]
@y3lousso That looks a bit different than what I saw in the linked SO answer :thinking: But I'm no expert with cron and BusyBox :grimacing:

**Comment by y3lousso** [comment]
@jlelse It is just another way to do it. Taken from [lobaro/restic-backup-docker](https://github.com/lobaro/restic-backup-docker).
The strange thing is that just removing go-cron from the Dockerfile building stage makes the container crash after the entrypoint.

**Comment by djmaze** [comment]
I don't really like the complicated setup and logging behaviour of the original cron. In our case, everything should happen in the foreground so it will be corretly logged by docker.

I thus went and [updated go-cron](https://github.com/djmaze/go-cron) to use the latest golang cron library. go-cron now works reliably for me, at least when testing manually. 

@jlelse @y3lousso You might want to test the image `mazzolino/restic:update-go-cron` which contains the changes from #12.

**Comment by y3lousso** [comment]
#12 works perfectly. Could you consider adding curl to the official image ? It would make it easier to use Slack notification via the POST_COMMANDS. Example : 
`curl -X POST -H 'Content-type: application/json' --data '{"text":"Backup was successfull !"}' https://hooks.slack.com/services/my_web_hook`

**Comment by djmaze** [comment]
> #12 works perfectly.

@y3lousso Great. I will make a new release then.

>  Could you consider adding curl to the official image ?

Would you mind creating a new issue for this?
